SuperblocksHQ/super-web3-provider
# Add tests to use-sinon branch

**[superblocks.provider]**

+ Add console log stub

+ Add a set of valid protocol tests around `isValidEndpoint`
+ Add a set of error checks against `isValidEndpoint`

+ Add test covering successful `init`
+ Add a set of tests covering possible `init` failures

+ Add test retrieving accounts successfully
+ Add test failing to retrieve accounts due to initialized provider

+ Add tests to cover `eth_accounts` via `sendMessage`
+ Add tests covering _API_ calls via `sendMessage`
+ Add tests covering _RPC_ calls via `sendMessage`

+ Add test to verify _eth_accounts_ via plain `send`
+ Add test to verify _API_ calls via `send`
+ Add test to verify _RPC_ calls via `send`

+ Add test to check _eth_accounts_ via direct `sendAsync` call
+ Add couple of tests checking _API_ calls via `sendAsync`
+ Add test to check _RPC_ calls via `sendAsync`

- Remove pending tests as they are now covered by this patch

- Remove commented out tests as they are covered by the additions

**[pusher.client]**
+ Add `console` `log` stub

* Organize tests in different `describe` blocks

- Remove other tests marked as pending as they are either not possible to test or they are now covered by this patch


**[rpc.client]**

+ Add test verifying successful _JSON-RPC_ request

+ Add test to check _JSON-RPC_ endpoint request failure


**[superblocks.client]**

+ Add `createRelease` test block

+ Add test to verify successful release creation

+ Add test to check release creation failure case

* Separate each set of tests in its own `describe` block

* Move `stub` code up a level

* Move `mockUtils` initialization up a level

* Rename error tests to clarify _API_ is accessible with an error
